### PR TITLE
GHC 8.10 support

### PR DESCRIPTION
**Pull request by sheaf**
A few changes to support GHC 8.10.

Relies on:
  - `haskell-lsp-types` PR https://github.com/alanz/haskell-lsp/pull/213
  - `hie-bios` PR https://github.com/mpickering/hie-bios/pull/134

### CODE REVIEW

**Comment by sheaf** [comment]
With this patch, `ghcide` works for me with `ghc-8.10.0.202001231` (GHC 8.10 rc1), and builds without warnings.

File: src/Development/IDE/GHC/Util.hs [hunk 1]
```diff
@@ -109,17 +113,22 @@ runGhcEnv env act = do
 -- | A 'DynFlags' value where most things are undefined. It's sufficient to call pretty printing,
 --   but not much else.
 fakeDynFlags :: DynFlags
-fakeDynFlags = defaultDynFlags settings mempty
+#if MIN_GHC_API_VERSION(8,10,0)
+fakeDynFlags = unsafeGlobalDynFlags
```
[the remark this answers is not in the document]
Comment (sheaf) [reply src/Development/IDE/GHC/Util.hs]: How would I check that? In VS Code for me ghcide works straight away, and the tests seem to run fine too (only 3 unexpected failures: cpp, plugins, TemplateHaskell).

File: src/Development/IDE/GHC/CPP.hs [hunk 2]
```diff
@@ -29,7 +29,7 @@ import DynFlags
 import Panic
 import FileCleanup
 #if MIN_GHC_API_VERSION(8,8,2)
-import LlvmCodeGen (LlvmVersion, llvmVersionList)
+import LlvmCodeGen (llvmVersionList)
```
[the remark this answers is not in the document]
Comment (sheaf) [reply src/Development/IDE/GHC/CPP.hs]: I can do that. GHC's `doCpp` function doesn't include `SysTools.Option "-D__GHCIDE__"]`, is that an issue?

[the remark this answers is not in the document]
Comment (ndmitchell) [reply src/Development/IDE/GHC/CPP.hs]: That should never have gone in this module - I've moved it in #377.

[the remark this answers is not in the document]
Comment (ndmitchell) [reply src/Development/IDE/GHC/CPP.hs]: This should never have gone here - I've moved it in #377

[the remark this answers is not in the document]
Comment (ndmitchell) [reply src/Development/IDE/GHC/CPP.hs]: Let's see - #378

File: src/Development/IDE/GHC/Util.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndmitchell) [reply src/Development/IDE/GHC/Util.hs]: Let's see - #378

**Comment by sheaf** [comment]
Hopefully `fakeDynFlags` won't be necessary in the future if a patch like [this one (Disentangle DynFlags and SDoc)](https://gitlab.haskell.org/ghc/ghc/merge_requests/2612) lands.

**Pull request merged**